addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxAudioSink.java

```diff
@@ -63,21 +69,26 @@ public String getLabel(Locale locale) {
     }
 
     @Override
-    public void process(AudioStream audioStream) throws UnsupportedAudioFormatException {
+    public void process(AudioStream audioStream)
+            throws UnsupportedAudioFormatException, UnsupportedAudioStreamException {
         AudioFormat format = audioStream.getFormat();
         if (!AudioFormat.WAV.isCompatible(format) && !AudioFormat.MP3.isCompatible(format)) {
             throw new UnsupportedAudioFormatException("Currently only MP3 and WAV formats are supported: ", format);
         }
 
         String url;
-        if (audioStream instanceof FixedLengthAudioStream) {
+        if (audioStream instanceof URLAudioStream) {
+            url = ((URLAudioStream) audioStream).getURL();
+        } else if (audioStream instanceof FixedLengthAudioStream) {
             // Since Squeezebox will make multiple requests for the stream, set a timeout on the stream
             url = audioHTTPServer.serve((FixedLengthAudioStream) audioStream, STREAM_TIMEOUT).toString();
 
             if (AudioFormat.WAV.isCompatible(format)) {
-                url += ".wav";
+                // TODO add FileAudioStream.EXTENSION_SEPARATOR
```

Comment:
@cweitkamp  Normally I would submit a PR to fix it, but I'm not able to do that until several days from now (no access to my dev environment).  Unfortunately, there are people who are unable to use TTS right now.  Can you fix it?